metadata-service/plugin/src/main/java/com/datahub/plugins/loader/PluginPermissionManagerImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.datahub.plugins.loader;
+
+import com.datahub.plugins.common.PluginPermissionManager;
+import com.datahub.plugins.common.SecurityMode;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.security.CodeSource;
+import java.security.Permissions;
+import java.security.ProtectionDomain;
+import java.security.cert.Certificate;
+
+
+public class PluginPermissionManagerImpl implements PluginPermissionManager {
+
+  private SecurityMode _securityMode;
+  public PluginPermissionManagerImpl(SecurityMode securityMode) {
```

Comment:
What happens if the constructor argument is null?

Reply:
In application.yaml , it is default set to `RESTRICTED`.

FYI... These modes are not working in this PR as spring boot fails if we enable the Java Security Manager. Detail of this issue can be found in https://linear.app/acryl-data/issue/ACR-3593/pluggable-authentication-authorizer

Reply:
So what is the plan here?

Reply:
Keep it as is in beta and allocate some time for me to debug and fix this part.

I have added notes in the permission section of plugin doc